src/cc_catalog_airflow/templates/template_provider.py_template

```diff
@@ -0,0 +1,374 @@
+    image_url, height, width = _get_image_info(media_data)
+    if image_url is None:
+        print (f"Found no media url:")
+        print(f"{json.dumps(media_data, indent=2)}")
+        check_and_save_json_for_test('no_image_url', media_data)
+        return None
+    item_license = _get_license(media_data)
+    if item_license is None:
+        print (f"Found no item license:")
+        print(f"{json.dumps(media_data, indent=2)}")
+        check_and_save_json_for_test('no_license', media_data)
+        return None
+    foreign_identifier = _get_foreign_identifier(media_data)
+    title = _get_title(media_data)
+    creator, creator_url = _get_creator_data(media_data)
+    thumbnail = _get_thumbnail_url(media_data)
+    metadata = _get_metadata(media_data)
+    tags = _get_tags(media_data)
+    check_and_save_json_for_test('full_item', media_data)
+
+    return {
+        'title': title,
+        'creator': creator,
+        'creator_url': creator_url,
+        'foreign_identifier': foreign_identifier,
+        'foreign_landing_url': foreign_landing_url,
+        'image_url': image_url,
+        'height': height,
+        'width': width,
+        'thumbnail_url': thumbnail,
+        'item_license': item_license,
+        'metadata': metadata,
+        'tags': tags
+    }
+
+
+def _get_foreign_identifier(media_data):
+    try:
+        return media_data['some_key'][0]['uid']
+    except(TypeError, IndexError, AttributeError):
+        return None
+
+
+def _get_image_info(media_data):
+    width = media_data.get('width')
+    height = media_data.get('height')
+    image_url = media_data.get('image_url')
+    return image_url, width, height
+
+def _get_audio_info(media_data):
+    duration = media_data.get('duration')
+    audio_url = media_data.get('audio_url')
+    return audio_url, duration
+
+
+def _get_thumbnail_url(media_data):
+    # TODO: Add correct implementation of _get_thumbnail_url
+    return media_data.get('thumbnail', {}).get('url', None)
+
+
+def _get_creator_data(item):
+    # TODO: Add correct implementation of _get_creator_data
+    creator = item.get('creator_key').strip()
+    creator_url = _cleanse_url(
+        item.get('creator_key', {}).get('url')
+    )
+    return creator, creator_url
+
+
+def _get_title(item):
+    # TODO: Add correct implementation of _get_title
+    title = item.get('title')
+    return title
+
+
+def _get_metadata(item):
+    """
+    Metadata may include: description, date created and modified at source,
+    categories, popularity statistics.
+    """
+    # TODO: Add function to extract metadata from the image_data dictionary
+    #  Do not includes keys without value
+    metadata = {}
+    some_other_key_value = item.get('some_other_key')
+    if some_other_key_value is not None:
+        metadata['some_other_key'] = some_other_key_value
+    return metadata
+
+
+def _get_tags(item):
+    """What tags do we save???"""
+    return item.get('tags')
```

Comment:
I see these tags (`raw_tags`) as keywords that describe or are related to the media item in the provider view. E.g. StockSnap have both, tags and keywords.